src/opt/memory_coloring_impl.cpp

```diff
@@ -185,8 +202,9 @@ void memory_coloring_impl::rewrite()
     std::vector<std::size_t> dims;
     dims.push_back((required_bytes + sizeof(float) - 1) / sizeof(float));
     shape s                       = {shape::float_type, dims};
-    instruction_ref scratch_param = p_program->add_parameter("scratch", s);
-    for(auto ins : iterator_for(*p_program))
+    std::string name              = p_mod->name() + "scratch";
```

Comment:
Why can't this just be `scratch`?

Reply:
the preallocation_param pass has an id "scratch", which is store per program. We need to use the mod name to differentiate the scratch for different modules.

Reply:
The preallocation_param pass should be updated to store the parameter for each module it is used in.

Reply:
fixed